### ISSUE TITLE
From #4. Interface webgazer and mouse-down events

### ISSUE THREAD

**Issue by jsimonson2013** [Madison-SDC/360WebViewer#14]
The webgazer API offers gaze position and 360-content requires mouse-down events for panning. Register gaze events within the 360-content as mouse-down events to continuously pan the scences.

**Comment by jsimonson2013** [comment]
From the WebGazer API Docs:

`{x:, y:} getCurrentPrediction()` 
`request a prediction from webgazer`
`returns object or null`
`x - predicted gaze x screen coordinate in pixels`
`y - predicted gaze y screen coordinate in pixels`

**Comment by jsimonson2013** [comment]
From stackoverflow post http://stackoverflow.com/questions/6157929/how-to-simulate-a-mouse-click-using-javascript

**Comment by jsimonson2013** [comment]
Seems like we could make calls to webgazer API to get x,y pos. If x,y pos is inside of our 360 content div, modify to have onmousedown = true.

**Comment by davidfei222** [comment]
Does this mean we're going to skip trying to get mouse movement input to work and go straight to eye movement?  Or are we still going to try to get mousing around to work first?  I feel like having mouse movement input would be a nice backup option to have in addition to the webgazer input.

**Comment by jsimonson2013** [comment]
Mouse movement works by modifying a-frame's code, which you can do in your project by commenting out line 58012 `window.addEventListener('mouseup', this.releaseMouse, false);` in aframe.js in libs. We may want to explore more into how other 360 content works for look controls.

**Comment by jsimonson2013** [comment]
https://github.com/alexrkass/aframe-no-click-look-controls an example of a custom look control

**Comment by jsimonson2013** [comment]
We should be able to do something very similar but use the gaze input instead of mouse

**Comment by davidfei222** [comment]
We've noticed that the gaze input at the moment is extremely jittery and not very accurate.  We tried using different trackers like tracker.js but it remains jittery.  Also, the webgazer input doesn't pan up or down very well (It only pans in a diagonal along the screen) --> Never mind, I figured out how to calibrate it so that it actually moves based on my eye movements.  However, it's still jittery as hell

**Comment by jsimonson2013** [comment]
Thanks for looking into it, we can modify the rate at which we poll to try to eliminate the saccades. There is literature on this type of approach and i think we could use a `wait ()` call to accomplish it. We will sacrifice some accuracy from more complex eye trackers and will have to approach it with a hacker mentality because of this

**Comment by jsimonson2013** [comment]
This describes how to write a custom component in a-frame which is what we will have to do to implement gaze control http://ngokevin.com/blog/aframe-component/

**Comment by jsimonson2013** [comment]
I think I need to better understand the build process of the library to implement a new control. The program references the built aframe.js and the current setup does not build a new library on running. This may mean we have to define a build process for the project that dynamically builds the libraries each time it is run.

**Comment by davidfei222** [comment]
So more problems regarding jitter: the webgazer positioning input starts off fairly accurate, but becomes progressively worse as the application runs to the point where the dot is literally flying all over the screen (even with calibration).  Zach and I tried experimenting with wait blocks in webgazer example, but we realized that the issue is with how webgazer itself collects data on the where the user is looking.  We tried looking through the webgazer library but have been unable to figure out how webgazer is collecting user input (the documentation is shit, it doesn't tell us anything we need to know).
